File: lab_05/CPP/TMatrix.cpp
#include "TMatrix.hpp"
#include <cstdio>


TMatrix::TMatrix(unsigned int i_dimension)
{
    dimension = i_dimension;
    matrix = new int *[dimension];

    for (unsigned int i = 0; i < dimension; ++i){
        matrix[i] = new int[dimension];
    }
}


TMatrix::TMatrix()
{
    dimension = 0;
    matrix = 0;
}


TMatrix::~TMatrix()
{
    for (unsigned int i = 0; i < dimension; ++i){
        if (matrix[i])
            delete [] matrix[i];
    }

    if (matrix)
        delete [] matrix;
}


void TMatrix::EnterMatrix()
{
    printf("Enter %d-dimension matrix row by row:\n", dimension);

    for (unsigned int i = 0; i < dimension; ++i){
        putchar('\t');
        for (unsigned int j = 0; j < dimension; ++j){
            scanf("%d", &matrix[i][j]);
        }
    }
    putchar('\n');
}


void TMatrix::PrintMatrix()
{
    for (unsigned int i = 0; i < dimension; ++i){
        putchar('\t');
        for (unsigned int j = 0; j < dimension; ++j){
            printf("%-5d ", matrix[i][j]);
        }
        putchar('\n');
    }
    putchar('\n');
}


int TMatrix::Sum() const
{
    int result = 0;
     
    for (unsigned int i = 0; i < dimension; ++i){
        for (unsigned int j = 0; j < dimension; ++j){
            result += matrix[i][j];
        }
    }
    return result;
}

File: lab_05/PY/TMatrix.py
class TMatrix():
    def __init__(self, dimension):
        self.dimension = dimension
        self.matrix = []

    def EnterMatrix(self):
        print(f"Enter {self.dimension} matrix row by row:")
        for i in range(self.dimension):
            self.matrix.append(list(map(int, input().split())))

    def PrintMatrix(self):
        for i in range(self.dimension):
            for j in range(self.dimension):
                print(self.matrix[i][j], end = " ")
            print()

    def Sum(self):
        result = 0
        for i in range(self.dimension):
            for j in range(self.dimension):
                result += self.matrix[i][j]
        return result

    def Det():
        pass